pull request to pass R CMD check and Travis CI

thierrygosselin opened the pull request:
* Removed the `misc` folder because you probably don’t want to show this to users
* Removed `require(“package_name”)` in various places because it shouldn't be use inside package. Instead, I used `::` and modified the `Depends` and `Imports` fields in the DESCRIPTION
* Added short name, title, description and keyword for the cpp function 
* Added `#' \dontrun{}` to `@examples` to go through R CMD check quickly
* Because package size was > 5 MB, I resized the figures in the vignettes and changed to `.PNG`. These modifications don't affect reading of the vignette. The package is now = 5 MB, this might throw some notes or warnings, if submitted to CRAN. Consequently, you might want to reduce the size of the data (use less markers maybe?)
* Added a few lines in the `.travis.yml` file to use remotes packages
* new version to trigger a Travis build

**Package now pass R CMD check and Travis CI**

Again... awesome work, congratulations to the R hackathon team!

caitiecollins commented:
Hi Thierry,

Wow. These look like great changes. Thanks very much!

Have we met, by the way? Did you come across MINOTAUR through NESCent or by
way of contact with other members of our group?

I'd like to accept your commit, especially because it looks like you put in
quite a bit of work combing through the code making useful changes (some of
which have been on my neglected to-do list!). I may need to first check
with the other co-contributors to the package, but are you interested in
continuing to contribute to the project?

I should also say that we have just submitted the main initial paper
describing the package, and this is still in the review process, so we have
forked the project onto a "develop" branch in addition to the "master"
branch. The reviewers have been directed to use the master branch (which we
have promised to keep as stable and un-changing as possible), so we are
trying to make all new changes on the develop branch (particularly changes
that might break the package or affect the performance/appearance of the
package). That said, it looks like the changes you have suggested are more
likely to improve than threaten the stability of the package, so for this
time I think it's fine if they are made to the master branch.

Thanks again for taking the time to make those changes. And please do tell
me a little about yourself, your interests, and whether you would like to
continue to contribute to MINOTAUR.

All the best,
Caitlin Collins.


Department of Infectious Disease Epidemiology
Imperial College London

On Thu, Jul 28, 2016 at 5:56 AM, Thierry Gosselin <notifications@github.com>
wrote:

>
>    - Removed the misc folder because you probably don’t want to show this
>    to users
>    - Removed require(“package_name”) in various places because it
>    shouldn't be use inside package. Instead, I used :: and modified the
>    Depends and Imports fields in the DESCRIPTION
>    - Added short name, title, description and keyword for the cpp
>    function
>    - Added #' \dontrun{} to @examples to go through R CMD check quickly
>    - Because package size was > 5 MB, I resized the figures in the
>    vignettes and changed to .PNG. These modifications don't affect
>    reading of the vignette. The package is now = 5 MB, this might throw some
>    notes or warnings, if submitted to CRAN. Consequently, you might want to
>    reduce the size of the data (use less markers maybe?)
>    - Added a few lines in the .travis.yml file to use remotes packages
>    - new version to trigger a Travis build
>
> *Package now pass R CMD check and Travis CI*
>
> Again... awesome work, congratulations to the R hackathon team!
> ------------------------------
> You can view, comment on, or merge this pull request online at:
>
>   https://github.com/NESCent/MINOTAUR/pull/9
> Commit Summary
>
>    - * deleted, not necessary
>    - * deleted, move outside package
>    - * added citation file
>    - * new fields and modified some fields to be more accurate with names
>    of authors and creator
>    - * replaced images in vignette for .png images of smaller size to
>    have a package less than 5 MB
>    - * new travis CI
>    - * new appveyor badge
>    - Included documentation
>    - * removed unnecessary parts
>    - encoding problem in the reference: non-ASCII
>    - * removed `require` that shouldn't be used inside package
>    - small edits
>    - * removed `require` that shouldn't be used inside package
>    - * Added `#' \dontrun{}` to `@examples` to go through R CMD check
>    quickly
>    - encoding problem with non-ASCII character
>    - * TRAVIS CI and APPVEYOR badge
>    - edit
>    - * new version to trigger a travis build
>    - * using `::` for external package dependencies
>    - * added just a tiny description of what the package does, based on
>    the shiny description provided
>    - * include `::` for packages dependencies
>    - * added `::` for required packages
>    - simple
>    - * numerous edit to add `::` to functions
>    - * numerous edit to pass: `R CMD check`
>    - test
>    - * remotes specification for Travis
>    - edit travis ci remotes
>    - * `github::Bioconductor-mirror/BiocInstaller` is required for
>    `github::Bioconductor-mirror/OmicCircos` to work...
>    - * edit remotes
>    - * removing APPVEYOR until further testing
>
> File Changes
>
>    - *M* .Rbuildignore
>    <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-0> (1)
>    - *M* .travis.yml
>    <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-1> (22)
>    - *M* DESCRIPTION
>    <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-2> (52)
>    - *D* DESCRIPTION~
>    <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-3> (29)
>    - *M* NAMESPACE
>    <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-4> (20)
>    - *D* NAMESPACE~
>    <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-5> (24)
>    - *M* R/MINOTAUR.R
>    <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-6> (200)
>    - *M* R/RcppExports.R
>    <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-7> (32)
>    - *M* R/data.R
>    <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-8> (3)
>    - *M* R/distanceFunctions.R
>    <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-9> (76)
>    - *M* R/plot_2D.R
>    <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-10> (64)
>    - *M* README.md
>    <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-11> (9)
>    - *A* inst/CITATION
>    <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-12> (13)
>    - *D* inst/mainserver/old_files/data.R
>    <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-13> (134)
>    - *D* inst/mainserver/old_files/server.R
>    <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-14> (274)
>    - *D* inst/mainserver/old_files/server_Manhattanplot.R
>    <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-15> (139)
>    - *D* inst/mainserver/old_files/server_circleplot.R
>    <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-16> (127)
>    - *D* inst/mainserver/old_files/server_cleanupData.R
>    <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-17> (934)
>    - *D* inst/mainserver/old_files/server_scatterplot1.R
>    <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-18> (197)
>    - *D* inst/mainserver/old_files/server_scatterplot2.R
>    <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-19> (187)
>    - *D* inst/mainserver/old_files/ui.R
>    <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-20> (696)
>    - *D* inst/mainserver/old_files/uiFunctions.R
>    <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-21> (527)
>    - *D* inst/mainserver/old_files/utils.R
>    <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-22> (167)
>    - *M* inst/mainserver/server.R
>    <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-23> (24)
>    - *M* inst/mainserver/server_calc_multivariate.R
>    <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-24> (490)
>    - *M* inst/mainserver/server_compare_outliers.R
>    <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-25> (23)
>    - *M* inst/mainserver/server_format_data.R
>    <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-26> (293)
>    - *M* inst/mainserver/server_input_data.R
>    <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-27> (443)
>    - *M* inst/mainserver/server_message.R
>    <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-28> (12)
>    - *M* inst/mainserver/server_plot_Manhattan.R
>    <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-29> (726)
>    - *M* inst/mainserver/server_plot_circularManhattan.R
>    <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-30> (820)
>    - *M* inst/mainserver/server_plot_histogram.R
>    <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-31> (280)
>    - *M* inst/mainserver/server_plot_scatterPlot.R
>    <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-32> (699)
>    - *M* inst/mainserver/server_welcome.R
>    <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-33> (84)
>    - *M* inst/mainserver/ui.R
>    <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-34> (407)
>    - *A* man/C_harmonicDist.Rd
>    <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-35> (13)
>    - *A* man/C_harmonicDist_partial.Rd
>    <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-36> (13)
>    - *A* man/C_kernelDeviance.Rd
>    <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-37> (13)
>    - *A* man/C_kernelDeviance_partial.Rd
>    <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-38> (13)
>    - *A* man/C_kernelDist.Rd
>    <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-39> (13)
>    - *A* man/C_kernelDist_partial.Rd
>    <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-40> (13)
>    - *A* man/C_neighborDist.Rd
>    <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-41> (13)
>    - *A* man/C_neighborDist_partial.Rd
>    <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-42> (13)
>    - *M* man/MINOTAUR.Rd
>    <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-43> (8)
>    - *M* man/Mahalanobis.Rd
>    <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-44> (6)
>    - *M* man/TwoRefSim.Rd
>    <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-45> (2)
>    - *M* man/harmonicDist.Rd
>    <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-46> (4)
>    - *M* man/kernelDeviance.Rd
>    <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-47> (4)
>    - *M* man/kernelDist.Rd
>    <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-48> (4)
>    - *M* man/neighborDist.Rd
>    <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-49> (4)
>    - *D* misc/CC_misc.R
>    <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-50> (12)
>    - *D* misc/DESCRIPTION_ori.txt
>    <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-51> (28)
>    - *D* misc/Deploy.R
>    <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-52> (76)
>    - *D* misc/Manuscript_draft.odt
>    <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-53> (0)
>    - *D* misc/colors/black_dashboardSidebar_#222D32.png
>    <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-54> (0)
>    - *D* misc/colors/black_light_dashboardSidebar_#2C3B41.png
>    <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-55> (0)
>    - *D* misc/colors/gray_blue_dashboard_background_#ECF0F5.png
>    <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-56> (0)
>    - *D* misc/colors/gray_welcome_#F5F5F5.png
>    <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-57> (0)
>    - *D* misc/colors/yellow_welcome_#FFFFCC.png
>    <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-58> (0)
>    - *D* misc/old_data/HumanGWAS.rda
>    <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-59> (0)
>    - *D* misc/old_data/NonParamEx.pdf
>    <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-60> (0)
>    - *D* misc/old_data/NonParamEx.rda
>    <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-61> (0)
>    - *D* misc/old_data/NonParamEx.txt
>    <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-62> (10002)
>    - *D* misc/old_data/NonParamEx1.Rdata
>    <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-63> (0)
>    - *D* misc/old_data/NonParamEx1.rda
>    <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-64> (0)
>    - *D* misc/old_data/NonParamEx2.Rdata
>    <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-65> (0)
>    - *D* misc/old_data/NonParamEx2.rda
>    <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-66> (0)
>    - *D* misc/old_data/NonParametric1.Rdata
>    <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-67> (0)
>    - *D* misc/old_data/NonParametric2.Rdata
>    <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-68> (0)
>    - *D* misc/old_data/NonParametricInverse.Rdata
>    <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-69> (0)
>    - *D* misc/old_data/NonParametricMultimodal.Rdata
>    <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-70> (0)
>    - *D* misc/old_data/OneRefSim.rda
>    <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-71> (0)
>    - *D* misc/old_data/OneRefSim.txt
>    <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-72> (1)
>    - *D* misc/old_data/TwoRefSim.Rdata
>    <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-73> (0)
>    - *D* misc/old_data/TwoRefSim.rda
>    <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-74> (0)
>    - *D* misc/old_data/TwoRefSim.txt
>    <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-75> (10001)
>    - *D* misc/old_data/TwoRefSimForShinyMCD.txt
>    <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-76> (10001)
>    - *D* misc/old_data/allanstrand-cline-parameters.csv
>    <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-77> (311)
>    - *D* misc/old_data/df_inverse.csv
>    <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-78> (1003)
>    - *D* misc/old_data/df_multimodal.csv
>    <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-79> (1001)
>    - *D* misc/old_data/largeData.csv
>    <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-80> (17625)
>    - *D* misc/old_data/largeData.rda
>    <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-81> (0)
>    - *D* misc/old_data/largeData.xls
>    <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-82> (0)
>    - *D* misc/old_data/largeData.xlsx
>    <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-83> (0)
>    - *D* misc/old_data/largeData2.ods
>    <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-84> (0)
>    - *D* misc/old_data/largeData2.xlsx
>    <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-85> (0)
>    - *D* misc/old_data/smallData.csv
>    <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-86> (35647)
>    - *D* misc/old_data/smallData.rda
>    <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-87> (0)
>    - *D* misc/old_data/smallData.xlsx
>    <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-88> (35647)
>    - *D* misc/startServer.R
>    <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-89> (28)
>    - *M* src/harmonicDist.cpp
>    <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-90> (26)
>    - *M* src/kernelDeviance.cpp
>    <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-91> (31)
>    - *M* src/kernelDist.cpp
>    <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-92> (30)
>    - *M* src/neighborDist.cpp
>    <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-93> (28)
>    - *D* vignettes/10_MultivariateTable.jpg
>    <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-94> (0)
>    - *A* vignettes/10_MultivariateTable.png
>    <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-95> (0)
>    - *D* vignettes/11_AxisOptions.jpg
>    <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-96> (0)
>    - *A* vignettes/11_AxisOptions.png
>    <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-97> (0)
>    - *D* vignettes/12_OutlierOptions.jpg
>    <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-98> (0)
>    - *A* vignettes/12_OutlierOptions.png
>    <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-99> (0)
>    - *D* vignettes/13_AesthOptions.jpg
>    <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-100> (0)
>    - *A* vignettes/13_AesthOptions.png
>    <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-101> (0)
>    - *D* vignettes/14_NdPlot.jpg
>    <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-102> (0)
>    - *A* vignettes/14_NdPlot.png
>    <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-103> (0)
>    - *D* vignettes/15_StackedPlots.jpg
>    <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-104> (0)
>    - *A* vignettes/15_StackedPlots.png
>    <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-105> (0)
>    - *D* vignettes/1_WelcomePage.jpg
>    <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-106> (0)
>    - *A* vignettes/1_WelcomePage.png
>    <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-107> (0)
>    - *D* vignettes/2_Menu.jpg
>    <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-108> (0)
>    - *A* vignettes/2_Menu.png
>    <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-109> (0)
>    - *D* vignettes/3_LoadDataMenu.jpg
>    <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-110> (0)
>    - *A* vignettes/3_LoadDataMenu.png
>    <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-111> (0)
>    - *D* vignettes/4_DataSummary.jpg
>    <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-112> (0)
>    - *A* vignettes/4_DataSummary.png
>    <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-113> (0)
>    - *D* vignettes/5_IdentifyPositions.jpg
>    <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-114> (0)
>    - *A* vignettes/5_IdentifyPositions.png
>    <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-115> (0)
>    - *D* vignettes/6_FilterDataOptions.jpg
>    <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-116> (0)
>    - *A* vignettes/6_FilterDataOptions.png
>    <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-117> (0)
>    - *D* vignettes/7_FilterDataSummaries.jpg
>    <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-118> (0)
>    - *A* vignettes/7_FilterDataSummaries.png
>    <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-119> (0)
>    - *D* vignettes/8_DistanceOptions.jpg
>    <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-120> (0)
>    - *A* vignettes/8_DistanceOptions.png
>    <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-121> (0)
>    - *D* vignettes/9_DensityOptions.jpg
>    <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-122> (0)
>    - *A* vignettes/9_DensityOptions.png
>    <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-123> (0)
>    - *M* vignettes/MINOTAUR.Rmd
>    <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-124> (32)
>
> Patch Links:
>
>    - https://github.com/NESCent/MINOTAUR/pull/9.patch
>    - https://github.com/NESCent/MINOTAUR/pull/9.diff
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/NESCent/MINOTAUR/pull/9>, or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AHRqq8DkSYYP8NYgz7Yr9Zk9XWjNeKwRks5qaDaCgaJpZM4JW3d7>
> .
>

thierrygosselin commented:
> 
> On Jul 28, 2016, at 13:22, caitiecollins <notifications@github.com> wrote:
> 
> Hi Thierry,
> 
> Wow. These look like great changes. Thanks very much!
> 
> Have we met, by the way? Did you come across MINOTAUR through NESCent or by
> way of contact with other members of our group?

Yes I was at the R hackathon and the editor for the MINOTAUR paper  for the special issue in MER ;)

Best
Thierry



> 
> I'd like to accept your commit, especially because it looks like you put in
> quite a bit of work combing through the code making useful changes (some of
> which have been on my neglected to-do list!). I may need to first check
> with the other co-contributors to the package, but are you interested in
> continuing to contribute to the project?
> 
> I should also say that we have just submitted the main initial paper
> describing the package, and this is still in the review process, so we have
> forked the project onto a "develop" branch in addition to the "master"
> branch.

The paper was accepted !


> The reviewers have been directed to use the master branch (which we
> have promised to keep as stable and un-changing as possible), so we are
> trying to make all new changes on the develop branch (particularly changes
> that might break the package or affect the performance/appearance of the
> package). That said, it looks like the changes you have suggested are more
> likely to improve than threaten the stability of the package, so for this
> time I think it's fine if they are made to the master branch.
> 
> Thanks again for taking the time to make those changes. And please do tell
> me a little about yourself, your interests, and whether you would like to
> continue to contribute to MINOTAUR.
> 
> All the best,
> Caitlin Collins.
> 
> 
> Department of Infectious Disease Epidemiology
> Imperial College London
> 
> On Thu, Jul 28, 2016 at 5:56 AM, Thierry Gosselin <notifications@github.com>
> wrote:
> 
> >
> > - Removed the misc folder because you probably don’t want to show this
> > to users
> > - Removed require(“package_name”) in various places because it
> > shouldn't be use inside package. Instead, I used :: and modified the
> > Depends and Imports fields in the DESCRIPTION
> > - Added short name, title, description and keyword for the cpp
> > function
> > - Added #' \dontrun{} to @examples to go through R CMD check quickly
> > - Because package size was > 5 MB, I resized the figures in the
> > vignettes and changed to .PNG. These modifications don't affect
> > reading of the vignette. The package is now = 5 MB, this might throw some
> > notes or warnings, if submitted to CRAN. Consequently, you might want to
> > reduce the size of the data (use less markers maybe?)
> > - Added a few lines in the .travis.yml file to use remotes packages
> > - new version to trigger a Travis build
> >
> > *Package now pass R CMD check and Travis CI*
> >
> > Again... awesome work, congratulations to the R hackathon team!
> > ------------------------------
> > You can view, comment on, or merge this pull request online at:
> >
> > https://github.com/NESCent/MINOTAUR/pull/9
> > Commit Summary
> >
> > - * deleted, not necessary
> > - * deleted, move outside package
> > - * added citation file
> > - * new fields and modified some fields to be more accurate with names
> > of authors and creator
> > - * replaced images in vignette for .png images of smaller size to
> > have a package less than 5 MB
> > - * new travis CI
> > - * new appveyor badge
> > - Included documentation
> > - * removed unnecessary parts
> > - encoding problem in the reference: non-ASCII
> > - * removed `require` that shouldn't be used inside package
> > - small edits
> > - * removed `require` that shouldn't be used inside package
> > - * Added `#' \dontrun{}` to `@examples` to go through R CMD check
> > quickly
> > - encoding problem with non-ASCII character
> > - * TRAVIS CI and APPVEYOR badge
> > - edit
> > - * new version to trigger a travis build
> > - * using `::` for external package dependencies
> > - * added just a tiny description of what the package does, based on
> > the shiny description provided
> > - * include `::` for packages dependencies
> > - * added `::` for required packages
> > - simple
> > - * numerous edit to add `::` to functions
> > - * numerous edit to pass: `R CMD check`
> > - test
> > - * remotes specification for Travis
> > - edit travis ci remotes
> > - * `github::Bioconductor-mirror/BiocInstaller` is required for
> > `github::Bioconductor-mirror/OmicCircos` to work...
> > - * edit remotes
> > - * removing APPVEYOR until further testing
> >
> > File Changes
> >
> > - *M* .Rbuildignore
> > <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-0> (1)
> > - *M* .travis.yml
> > <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-1> (22)
> > - *M* DESCRIPTION
> > <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-2> (52)
> > - *D* DESCRIPTION~
> > <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-3> (29)
> > - *M* NAMESPACE
> > <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-4> (20)
> > - *D* NAMESPACE~
> > <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-5> (24)
> > - *M* R/MINOTAUR.R
> > <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-6> (200)
> > - *M* R/RcppExports.R
> > <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-7> (32)
> > - *M* R/data.R
> > <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-8> (3)
> > - *M* R/distanceFunctions.R
> > <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-9> (76)
> > - *M* R/plot_2D.R
> > <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-10> (64)
> > - *M* README.md
> > <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-11> (9)
> > - *A* inst/CITATION
> > <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-12> (13)
> > - *D* inst/mainserver/old_files/data.R
> > <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-13> (134)
> > - *D* inst/mainserver/old_files/server.R
> > <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-14> (274)
> > - *D* inst/mainserver/old_files/server_Manhattanplot.R
> > <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-15> (139)
> > - *D* inst/mainserver/old_files/server_circleplot.R
> > <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-16> (127)
> > - *D* inst/mainserver/old_files/server_cleanupData.R
> > <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-17> (934)
> > - *D* inst/mainserver/old_files/server_scatterplot1.R
> > <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-18> (197)
> > - *D* inst/mainserver/old_files/server_scatterplot2.R
> > <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-19> (187)
> > - *D* inst/mainserver/old_files/ui.R
> > <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-20> (696)
> > - *D* inst/mainserver/old_files/uiFunctions.R
> > <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-21> (527)
> > - *D* inst/mainserver/old_files/utils.R
> > <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-22> (167)
> > - *M* inst/mainserver/server.R
> > <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-23> (24)
> > - *M* inst/mainserver/server_calc_multivariate.R
> > <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-24> (490)
> > - *M* inst/mainserver/server_compare_outliers.R
> > <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-25> (23)
> > - *M* inst/mainserver/server_format_data.R
> > <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-26> (293)
> > - *M* inst/mainserver/server_input_data.R
> > <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-27> (443)
> > - *M* inst/mainserver/server_message.R
> > <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-28> (12)
> > - *M* inst/mainserver/server_plot_Manhattan.R
> > <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-29> (726)
> > - *M* inst/mainserver/server_plot_circularManhattan.R
> > <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-30> (820)
> > - *M* inst/mainserver/server_plot_histogram.R
> > <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-31> (280)
> > - *M* inst/mainserver/server_plot_scatterPlot.R
> > <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-32> (699)
> > - *M* inst/mainserver/server_welcome.R
> > <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-33> (84)
> > - *M* inst/mainserver/ui.R
> > <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-34> (407)
> > - *A* man/C_harmonicDist.Rd
> > <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-35> (13)
> > - *A* man/C_harmonicDist_partial.Rd
> > <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-36> (13)
> > - *A* man/C_kernelDeviance.Rd
> > <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-37> (13)
> > - *A* man/C_kernelDeviance_partial.Rd
> > <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-38> (13)
> > - *A* man/C_kernelDist.Rd
> > <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-39> (13)
> > - *A* man/C_kernelDist_partial.Rd
> > <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-40> (13)
> > - *A* man/C_neighborDist.Rd
> > <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-41> (13)
> > - *A* man/C_neighborDist_partial.Rd
> > <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-42> (13)
> > - *M* man/MINOTAUR.Rd
> > <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-43> (8)
> > - *M* man/Mahalanobis.Rd
> > <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-44> (6)
> > - *M* man/TwoRefSim.Rd
> > <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-45> (2)
> > - *M* man/harmonicDist.Rd
> > <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-46> (4)
> > - *M* man/kernelDeviance.Rd
> > <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-47> (4)
> > - *M* man/kernelDist.Rd
> > <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-48> (4)
> > - *M* man/neighborDist.Rd
> > <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-49> (4)
> > - *D* misc/CC_misc.R
> > <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-50> (12)
> > - *D* misc/DESCRIPTION_ori.txt
> > <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-51> (28)
> > - *D* misc/Deploy.R
> > <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-52> (76)
> > - *D* misc/Manuscript_draft.odt
> > <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-53> (0)
> > - *D* misc/colors/black_dashboardSidebar_#222D32.png
> > <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-54> (0)
> > - *D* misc/colors/black_light_dashboardSidebar_#2C3B41.png
> > <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-55> (0)
> > - *D* misc/colors/gray_blue_dashboard_background_#ECF0F5.png
> > <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-56> (0)
> > - *D* misc/colors/gray_welcome_#F5F5F5.png
> > <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-57> (0)
> > - *D* misc/colors/yellow_welcome_#FFFFCC.png
> > <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-58> (0)
> > - *D* misc/old_data/HumanGWAS.rda
> > <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-59> (0)
> > - *D* misc/old_data/NonParamEx.pdf
> > <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-60> (0)
> > - *D* misc/old_data/NonParamEx.rda
> > <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-61> (0)
> > - *D* misc/old_data/NonParamEx.txt
> > <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-62> (10002)
> > - *D* misc/old_data/NonParamEx1.Rdata
> > <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-63> (0)
> > - *D* misc/old_data/NonParamEx1.rda
> > <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-64> (0)
> > - *D* misc/old_data/NonParamEx2.Rdata
> > <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-65> (0)
> > - *D* misc/old_data/NonParamEx2.rda
> > <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-66> (0)
> > - *D* misc/old_data/NonParametric1.Rdata
> > <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-67> (0)
> > - *D* misc/old_data/NonParametric2.Rdata
> > <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-68> (0)
> > - *D* misc/old_data/NonParametricInverse.Rdata
> > <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-69> (0)
> > - *D* misc/old_data/NonParametricMultimodal.Rdata
> > <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-70> (0)
> > - *D* misc/old_data/OneRefSim.rda
> > <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-71> (0)
> > - *D* misc/old_data/OneRefSim.txt
> > <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-72> (1)
> > - *D* misc/old_data/TwoRefSim.Rdata
> > <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-73> (0)
> > - *D* misc/old_data/TwoRefSim.rda
> > <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-74> (0)
> > - *D* misc/old_data/TwoRefSim.txt
> > <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-75> (10001)
> > - *D* misc/old_data/TwoRefSimForShinyMCD.txt
> > <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-76> (10001)
> > - *D* misc/old_data/allanstrand-cline-parameters.csv
> > <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-77> (311)
> > - *D* misc/old_data/df_inverse.csv
> > <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-78> (1003)
> > - *D* misc/old_data/df_multimodal.csv
> > <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-79> (1001)
> > - *D* misc/old_data/largeData.csv
> > <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-80> (17625)
> > - *D* misc/old_data/largeData.rda
> > <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-81> (0)
> > - *D* misc/old_data/largeData.xls
> > <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-82> (0)
> > - *D* misc/old_data/largeData.xlsx
> > <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-83> (0)
> > - *D* misc/old_data/largeData2.ods
> > <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-84> (0)
> > - *D* misc/old_data/largeData2.xlsx
> > <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-85> (0)
> > - *D* misc/old_data/smallData.csv
> > <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-86> (35647)
> > - *D* misc/old_data/smallData.rda
> > <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-87> (0)
> > - *D* misc/old_data/smallData.xlsx
> > <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-88> (35647)
> > - *D* misc/startServer.R
> > <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-89> (28)
> > - *M* src/harmonicDist.cpp
> > <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-90> (26)
> > - *M* src/kernelDeviance.cpp
> > <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-91> (31)
> > - *M* src/kernelDist.cpp
> > <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-92> (30)
> > - *M* src/neighborDist.cpp
> > <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-93> (28)
> > - *D* vignettes/10_MultivariateTable.jpg
> > <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-94> (0)
> > - *A* vignettes/10_MultivariateTable.png
> > <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-95> (0)
> > - *D* vignettes/11_AxisOptions.jpg
> > <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-96> (0)
> > - *A* vignettes/11_AxisOptions.png
> > <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-97> (0)
> > - *D* vignettes/12_OutlierOptions.jpg
> > <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-98> (0)
> > - *A* vignettes/12_OutlierOptions.png
> > <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-99> (0)
> > - *D* vignettes/13_AesthOptions.jpg
> > <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-100> (0)
> > - *A* vignettes/13_AesthOptions.png
> > <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-101> (0)
> > - *D* vignettes/14_NdPlot.jpg
> > <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-102> (0)
> > - *A* vignettes/14_NdPlot.png
> > <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-103> (0)
> > - *D* vignettes/15_StackedPlots.jpg
> > <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-104> (0)
> > - *A* vignettes/15_StackedPlots.png
> > <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-105> (0)
> > - *D* vignettes/1_WelcomePage.jpg
> > <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-106> (0)
> > - *A* vignettes/1_WelcomePage.png
> > <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-107> (0)
> > - *D* vignettes/2_Menu.jpg
> > <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-108> (0)
> > - *A* vignettes/2_Menu.png
> > <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-109> (0)
> > - *D* vignettes/3_LoadDataMenu.jpg
> > <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-110> (0)
> > - *A* vignettes/3_LoadDataMenu.png
> > <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-111> (0)
> > - *D* vignettes/4_DataSummary.jpg
> > <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-112> (0)
> > - *A* vignettes/4_DataSummary.png
> > <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-113> (0)
> > - *D* vignettes/5_IdentifyPositions.jpg
> > <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-114> (0)
> > - *A* vignettes/5_IdentifyPositions.png
> > <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-115> (0)
> > - *D* vignettes/6_FilterDataOptions.jpg
> > <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-116> (0)
> > - *A* vignettes/6_FilterDataOptions.png
> > <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-117> (0)
> > - *D* vignettes/7_FilterDataSummaries.jpg
> > <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-118> (0)
> > - *A* vignettes/7_FilterDataSummaries.png
> > <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-119> (0)
> > - *D* vignettes/8_DistanceOptions.jpg
> > <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-120> (0)
> > - *A* vignettes/8_DistanceOptions.png
> > <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-121> (0)
> > - *D* vignettes/9_DensityOptions.jpg
> > <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-122> (0)
> > - *A* vignettes/9_DensityOptions.png
> > <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-123> (0)
> > - *M* vignettes/MINOTAUR.Rmd
> > <https://github.com/NESCent/MINOTAUR/pull/9/files#diff-124> (32)
> >
> > Patch Links:
> >
> > - https://github.com/NESCent/MINOTAUR/pull/9.patch
> > - https://github.com/NESCent/MINOTAUR/pull/9.diff
> >
> > —
> > You are receiving this because you are subscribed to this thread.
> > Reply to this email directly, view it on GitHub
> > <https://github.com/NESCent/MINOTAUR/pull/9>, or mute the thread
> > <https://github.com/notifications/unsubscribe-auth/AHRqq8DkSYYP8NYgz7Yr9Zk9XWjNeKwRks5qaDaCgaJpZM4JW3d7>
> > .
> >
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub <https://github.com/NESCent/MINOTAUR/pull/9#issuecomment-235964235>, or mute the thread <https://github.com/notifications/unsubscribe-auth/ACC8Ir8UmIOQ5CAtWmxQ0KUsuWagruS9ks5qaOVwgaJpZM4JW3d7>.
>

caitiecollins commented:
Hi, 

Ok, well that makes sense. I'm sorry I wasn't sure if we had met. Now that you mention the cold, I do think I remember! 

I'm really glad to hear the paper was accepted. That's excellent news! 

I looked over the changes and they look good to me (I've merged them in now). I'm new to using travis and I just added it to the package, so I'm very grateful for the helping hand from someone more familiar with it than I. 

Many thanks, 
Caitlin.
